src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -751,11 +751,20 @@ AppendShardIdToName(char **name, uint64 shardId)
 				 SHARD_NAME_SEPARATOR, longNameHash,
 				 shardIdAndSeparator);
 	}
-	extendedNameLength = strlen(extendedName) + 1;
-	Assert(extendedNameLength <= NAMEDATALEN);
 
-	(*name) = (char *) repalloc((*name), extendedNameLength);
-	snprintf((*name), extendedNameLength, "%s", extendedName);
+	(*name) = (char *) repalloc((*name), NAMEDATALEN);
+	writtenBytes = snprintf((*name), NAMEDATALEN, "%s", extendedName);
```

Comment:
naming suggest it was returning actually written bytes, but it is not.
something along neededBytes, actualLength, requestedBytes ...  would look better